src/components/location/location.service.ts

```diff
@@ -109,7 +109,20 @@ export class LocationService {
       return [];
     }
     const createdAt = DateTime.local();
-    const propLabel = prop === 'name' ? 'Property:LocationName' : 'Property';
+    let propLabel;
+
+    if (prop === 'name') {
+      if (baseNode === 'newZone') {
+        propLabel = 'Property:FieldZoneName';
+      } else if (baseNode === 'newRegion') {
+        propLabel = 'Property:FieldRegionName';
+      } else {
+        propLabel = 'Property:LocationName';
+      }
+    } else {
+      propLabel = 'Property';
+    }
```

Comment:
This is a code smell. The extra labels should just be passed optionally passed in. 

```tsx
  property = (prop: string, value: any, baseNode: string, extraLabels?: string[]) => {
    const labels = ['Property', ...extraLabels];
```